src/ai/backend/manager/models/kernel.py

```diff
@@ -589,19 +589,24 @@ async def get_kernel(
                 raise SessionNotFound
             return cand[0]
 
-    @staticmethod
+    @classmethod
     async def set_kernel_status(
+        cls,
         db: ExtendedAsyncSAEngine,
         kernel_id: KernelId,
         status: KernelStatus,
         *,
         status_data: Optional[Mapping[str, Any]] = None,
         reason: Optional[str] = None,
+        current_time: Optional[datetime] = None,
```

Comment:
How about changing parameter name to `status_changed_at`? Updating "current time" does not make sense literally.
```suggestion
        status_changed_at: Optional[datetime] = None,
```